### PR TITLE
Weblab filesapi rename case only

### PR DESCRIPTION
**Pull request by cpirich**
* Handle case-only renames (e.g. `foo.html` to `Foo.html`). Cleaned up variable names in`files_put_file` relating to the 3 filenames in play (the new file, the source file, and the file to be deleted). It is now more clear which are unescaped and/or downcased.
* Add a new test for `case_only_rename`

### CODE REVIEW

File: shared/middleware/files_api.rb [hunk 1]
```diff
@@ -486,25 +486,35 @@ def copy_file(endpoint, encrypted_channel_id, filename, source_filename)
...
     # store the new file
-    if params['src']
-      new_entry_json = copy_file('files', encrypted_channel_id, downcased_filename, params['src'].downcase)
+    if unescaped_src_filename_downcased
+      if case_only_rename
+        src_entry = manifest.detect {|e| e['filename'].downcase == unescaped_filename_downcased}
+        not_found if src_entry.nil?
+        new_entry_json = src_entry.to_json
+      else
+        new_entry_json = copy_file('files', encrypted_channel_id, URI.encode(unescaped_filename_downcased), URI.encode(unescaped_src_filename_downcased))
```
Comment (ashercodeorg) [inline shared/middleware/files_api.rb:508]: nit: This line is a too long, please break it up as
```
new_entry_json = copy_file(
  'files',
  encrypted_channel_id,
  URI.encode(unescaped_filename_downcased),
  URI.encode(unescaped_src_filename_downcased)
)
```

File: shared/middleware/files_api.rb [hunk 2]
```diff
@@ -531,7 +541,7 @@ def files_put_file(encrypted_channel_id, filename, body)
     end
 
     # delete a file if requested (same as src file in a rename operation)
-    bucket.delete(encrypted_channel_id, params['delete'].downcase) if params['delete']
+    bucket.delete(encrypted_channel_id, URI.encode(unescaped_delete_filename_downcased)) if deleting_separate_file
```
Comment (ashercodeorg) [inline shared/middleware/files_api.rb:544]: nit: Another long line.
```
if deleting_separate_file
  bucket.delete(encrypted_channel_id, URI.encode(unescaped_delete_filename_downcased))
end
```

File: shared/test/test_files.rb [hunk 3]
```diff
@@ -425,6 +425,64 @@ def test_rename_mixed_case
...
+    assert_equal image_body, last_response.body
+
+    @api.get_object(escaped_filename2)
+    assert successful?
+    assert_equal image_body, last_response.body
+
+    get "v3/files/#{@channel_id}"
+    response_after_rename = JSON.parse(last_response.body)
+    # There should be only one file with the new filename, category, and size matching our expectations
+    expected_image_info_after_rename = {'filename' => filename2, 'category' => 'image', 'size' => image_body.length}
```
Comment (ashercodeorg) [inline shared/test/test_files.rb:471]: Is this clearer as `expected_image_info_after_rename = expected_image_info.merge('filename' => filename2)`?

File: shared/test/test_files.rb [hunk 4]
```diff
@@ -425,6 +425,64 @@ def test_rename_mixed_case
...
+    file_infos_after_rename = response_after_rename['files']
+    assert_equal(1, file_infos_after_rename.length)
+    assert_fileinfo_equal(expected_image_info_after_rename, file_infos_after_rename[0])
+
+    # The manifest version (filesVersionId) should be different, but the file version should be the same
+    assert response_before_rename['filesVersionId'] != response_after_rename['filesVersionId']
```
Comment (ashercodeorg) [inline shared/test/test_files.rb:477]: Perhaps use `refute_equal` here, as doing so may provide a more helpful error message should the test fail here.

**Comment by cpirich** [comment]
@ashercodeorg I'll make those changes locally and include in my next PR. Thanks for the feedback.

**Pull request merged**